numpy/lib/tests/test_mixins.py

```diff
@@ -0,0 +1,207 @@
+        check(divmod(2, ArrayLike(2)))
+
+        check(divmod(ArrayLike(2), np.array(2)))
+        check(divmod(np.array(2), ArrayLike(2)))
+
+        check(divmod(ArrayLike(np.array(2)), 2))
+        check(divmod(2, ArrayLike(np.array(2))))
+
+        check(divmod(ArrayLike(np.array(2)), np.array(2)))
+        check(divmod(np.array(2), ArrayLike(np.array(2))))
+
+    def test_inplace(self):
+        array_like = ArrayLike(np.array([0]))
+        array_like += 1
+        _assert_equal_type_and_value(array_like, ArrayLike(np.array([1])))
+
+        array = np.array([0])
+        array += ArrayLike(1)
+        _assert_equal_type_and_value(array, ArrayLike(np.array([1])))
+
+    def test_opt_out(self):
+
+        class OptOut(object):
+            """Object that opts out of __array_ufunc__."""
+            __array_ufunc__ = None
+
+            def __add__(self, other):
+                return self
+
+            def __radd__(self, other):
+                return self
+
+        array_like = ArrayLike(1)
+        opt_out = OptOut()
+
+        # supported operations
+        assert_(array_like + opt_out is opt_out)
+        assert_(opt_out + array_like is opt_out)
+
+        # not supported
+        with assert_raises(TypeError):
+            # don't use the Python default, array_like = array_like + opt_out
+            array_like += opt_out
+        with assert_raises(TypeError):
+            array_like - opt_out
+        with assert_raises(TypeError):
+            opt_out - array_like
+
+    def test_subclass(self):
+
+        class SubArrayLike(ArrayLike):
+            """Should take precedence over ArrayLike."""
+
+        x = ArrayLike(0)
+        y = SubArrayLike(1)
+        _assert_equal_type_and_value(x + y, y)
+        _assert_equal_type_and_value(y + x, y)
+
+    def test_unary_methods(self):
+        array = np.array([-1, 0, 1, 2])
+        array_like = ArrayLike(array)
+        for op in [operator.neg, operator.pos, abs, operator.invert]:
+            _assert_equal_type_and_value(op(array_like), ArrayLike(op(array)))
+
+    def test_binary_methods(self):
+        array = np.array([-1, 0, 1, 2])
+        array_like = ArrayLike(array)
+        operators = [
+            operator.lt,
+            operator.le,
+            operator.eq,
+            operator.ne,
+            operator.gt,
+            operator.ge,
+            operator.add,
+            operator.sub,
+            operator.mul,
+            operator.truediv,
+            operator.floordiv,
+            operator.mod,
+            # divmod is handled separately above
+            pow,
+            operator.lshift,
+            operator.rshift,
+            operator.and_,
+            operator.xor,
+            operator.or_,
+        ]
+        if PY2:
+            operators.append(operator.div)
+
+        with warnings.catch_warnings():
+            # ignore warnings from operator.div when run with python -3
+            warnings.filterwarnings('ignore', 'classic int division',
```

Comment:
This didn't work:
https://travis-ci.org/shoyer/numpy/jobs/221437434
https://travis-ci.org/shoyer/numpy/jobs/221437437

Any other ideas for how to ignore these warnings on Python 2?